### PR TITLE
WT-9755 Retain database files after a failure in run_format_configs.sh

### PR DESCRIPTION
**Pull request by ruby-oujo**
The RUNDIR (renamed WT_TEST in this PR) gets replaced by each new iteration of format config test runs, even when there is a failure. 

To preserve the database files when there is a failure for debugging purposes; we move the WT_TEST directory to a new directory if there is a failure and prevent it from being overwritten or deleted. 

The directory is named after the failure config in the format WT_TEST_{CONFIG}.

### CODE REVIEW

**Review by quchenhao — APPROVED** [review]
lgtm

File: test/evergreen/run_format_configs.sh [hunk 1]
```diff
@@ -10,7 +10,7 @@ set -e
 # Switch to the Git repo toplevel directory
 cd $(git rev-parse --show-toplevel)
 # Walk into the test/format directory
-cd cmake_build/test/format
+cd build/test/format
```
Comment (korteland) [inline test/evergreen/run_format_configs.sh:13]: this looks like a testing change that accidentally got included?

Comment (ruby-oujo) [reply test/evergreen/run_format_configs.sh:13]: Yes this is true, can change it back.

File: test/evergreen/run_format_configs.sh [hunk 2]
```diff
@@ -26,11 +26,12 @@ failure=0
...
 do
 	echo -e "\nTesting CONFIG $config ...\n"
-	if (./t -c $config); then
+	if (./t -c $config -h WT_TEST); then
 		let "success++"
 	else
 		let "failure++"
-		[ -f RUNDIR/CONFIG ] && cat RUNDIR/CONFIG
+		[ -f WT_TEST/CONFIG ] && cat WT_TEST/CONFIG
+		mv WT_TEST WT_TEST_${config##*/}
```
Comment (ajmorton) [inline test/evergreen/run_format_configs.sh:34]: Small tweak: We can strip the parent folders from `config` using `basename`.

I'd change 
`mv WT_TEST WT_TEST_${config##*/}`
to either 
`mv WT_TEST WT_TEST_$(basename $config)`
or
```
config_name=$(basename $config)
mv WT_TEST WT_TEST_${config_name}    
```

**Review by ajmorton — APPROVED** [review]
Looking good!
One recommendation but feel free to leave things as they are.

File: test/evergreen/run_format_configs.sh (hunk 2 above)
Comment (etienneptl) [reply test/evergreen/run_format_configs.sh:34]: And we could probably make `WT_TEST` a variable :)

**Pull request merged**